### PR TITLE
fix(api): enable stall detection

### PR DESCRIPTION
**Pull request by ahiuchingau**
# Overview
We previously disable stall detection because the Y axes were unreliable and used to skip steps often on our EVT models. Now that we have the DVT units, we should verify that the motors are accurate enough for us to safely re-enable stall detection.

On top of that, how do we feel about adding a robot feature flag to disable & enable this feature for testing purposes? 

## Changes:
* using new speeds and accelerations and new run current for Z/A from testing (data can be found [here](https://opentrons.atlassian.net/wiki/spaces/RPDO/pages/3516235793/Maximum+acceleration+and+speed+testing+DVT+OT-3))
* making sure stall check is disabled during tip pick up as stalling is expected during force pickup
* add a new feature flag to allow stall detection to be disabled

## Test Plan
In order to merge this PR, we need to put the changes in this PR on DVT several units (ideally, as many as we can) and make sure we can run a protocol or two without triggering any stall detection error.

- [x] Run on all DVT robots in 528
- [ ] DVT robots in 520

### CODE REVIEW

**Comment by sfoster1** [comment]
I think we shouldn't target this on the internal release if it needs that much testing, unfortunately. Let's put it on edge, though!

And I think in terms of testing, you're right on about how to do it. I think a feature flag is a pretty good idea, although it's sort of getting to the point where we should add mechanical revisions or something to handle this.

File: api/src/opentrons/config/advanced_settings.py [hunk 1]
```diff
@@ -181,6 +181,11 @@ class Setting(NamedTuple):
         title="Enable robots with the new usb connected rear-panel board.",
         description="This is an Opentrons-internal setting to test new rear-panel.",
     ),
+    SettingDefinition(
+        _id="disableStallDetection",
+        title="Disable stall detection on the OT-3.",
```
Comment (sfoster1) [inline api/src/opentrons/config/advanced_settings.py]:
```suggestion
        title="Disable stall detection on the Flex.",
```
(it's customer facing)

File: api/src/opentrons/config/defaults_ot3.py [hunk 2]
```diff
@@ -76,16 +76,16 @@
 DEFAULT_MAX_SPEEDS: Final[ByGantryLoad[Dict[OT3AxisKind, float]]] = ByGantryLoad(
     high_throughput={
         OT3AxisKind.X: 500,
-        OT3AxisKind.Y: 500,
+        OT3AxisKind.Y: 375,
```
Comment (sfoster1) [inline api/src/opentrons/config/defaults_ot3.py]: imo this should be a separate PR

File: api/src/opentrons/config/feature_flags.py [hunk 3]
```diff
@@ -35,3 +35,7 @@ def rear_panel_integration() -> bool:
     """Whether to enable usb connected rear_panel for the OT-3."""
 
     return advs.get_setting_with_env_overload("rearPanelIntegration")
+
+
+def disable_stall_detection() -> bool:
```
Comment (sfoster1) [inline api/src/opentrons/config/feature_flags.py]: kinda sounds like a verb-phrase, can we call this like `stall_detection_disabled()`

Comment (sfoster1) [reply api/src/opentrons/config/feature_flags.py]: or even `stall_detection_enabled()` and return `not get_setting...("disableStallDetection")`

**Review by sfoster1 — APPROVED** [review]
Got some minor feedback things but overall looks good as long as the hardware works now

**Review by nusrat813 — APPROVED** [review]
Tested, LGTM

**Review by skowalski08 — COMMENTED** [review]
tested 5/9. LGTM
